lighter/utils/model.py

```diff
@@ -53,14 +56,73 @@ def replace_layer_with_identity(model: Module, layer_name: str) -> Module:
     return replace_layer_with(model, layer_name, Identity())
 
 
-def remove_last_layer_sequentially(model: Module()) -> Sequential:
-    """Removes the last layer of a network and returns it as an nn.Sequential model.
-    Useful when a network is to be used as a backbone of an SSL model.
+def remove_n_last_layers_sequentially(model: Module(), num_layers=1) -> Sequential:
+    """Remove a number of last layers of a network and return it as an nn.Sequential model.
+    Useful when a network is used as a backbone of an SSL model.
 
     Args:
         model (Module): PyTorch model object.
+        num_layers (int, optional): Number of last layers to be removed. Defaults to 1.
 
     Returns:
         Sequential: PyTorch Sequential model with the last layer removed.
     """
-    return Sequential(*list(model.children())[:-1])
+    return Sequential(*list(model.children())[:-num_layers])
+
+
+def adjust_prefix_and_load_state_dict(model: Module, ckpt_path: str, ckpt_to_model_prefix: Dict[str, str] = None) -> Module:
+    """Load state_dict from a checkpoint into a model using `torch.load(strict=False`).
+    `ckpt_to_model_prefix` mapping allows to rename the prefix of the checkpoint's state_dict keys
+    so that they match those of the model's state_dict. This is often needed when a model was trained
+    as a backbone of another model, so its state_dict keys won't be the same to those of a standalone
+    version of that model. Prior to defining the `ckpt_to_model_prefix`, it is advised to manually check
+    for mismatch between the names and specify them accordingly.
+
+    Args:
+        model (Module): The PyTorch model instance to load the state_dict into.
+        ckpt_path (str): Path to the checkpoint.
+        ckpt_to_model_prefix (Dict[str, str], optional): A dictionary that maps keys in the checkpoint's
+            state_dict to keys in the model's state_dict. If None, no key mapping is performed. Defaults to None.
+
+    Returns:
+        Module: The model instance with the state_dict loaded.
+
+    Raises:
+        ValueError: If there is no overlap between checkpoint's and model's state_dict.
+    """
+
+    # Load checkpoint
+    ckpt = torch.load(ckpt_path)
```

Comment:
What if a `pathlib.Path`is passed? Will it still work?

Reply:
Yes. Do you think we should add https://docs.python.org/3/library/os.html#os.PathLike everywhere?

Reply:
Not necessarily. I think being aware that a `Path` could be passed as well is enough. Maybe explicitly check for `str` if needed. Some libraries, like SimpleITK do not work with `Path` but over time this will be a less and less frequent problem